appendices/spirvenv.txt

```diff
@@ -614,6 +614,11 @@ ifndef::VK_VERSION_1_3,VK_KHR_format_feature_flags2[]
     operand of 2 and an "`Image Format`" operand of code:Unknown must: be
     decorated with code:NonReadable.
 endif::VK_VERSION_1_3,VK_KHR_format_feature_flags2[]
+  * code:OpImageWrite to any variable created with a code:Type of
+    code:OpTypeImage that has a code:Sampled operand of 2 must: have the
+    code:Texel operand contain at least as many components as the
+    code:Image code:Format according to the
```

Comment:
It's not immediately clear to me how we would determine the number of components "according to the table". I expected to see a component count in there but there isn't one. (It's not too hard to work out what the correct answer is, but it would be nice to be more explicit here).

Reply:
So luckily the `vk.xml` has all this information now (which these tables should be generated from in the spec too), an example

```xml
        <format name="VK_FORMAT_R8G8_UINT" class="16-bit" blockSize="2" texelsPerBlock="1">
            <component name="R" bits="8" numericFormat="UINT"/>
            <component name="G" bits="8" numericFormat="UINT"/>
            <spirvimageformat name="Rg8ui"/>
        </format>
 ```
 
 So it will be possible to generate this check in the validation layers to map `Rg8ui` in SPIR-V to having 2 `<component>` tags

Reply:
My concern was more for someone reading the spec, rather than getting this right in the validation layers (although having this autogenerated there is definitely a good thing). My first thought is that this should say something like "at least as many components as are present in the corresponding Vulkan format, as given in table ...". That doesn't work for Unknown though, for which the corresponding format is "Any".

Reply:
I plan to take on https://github.com/KhronosGroup/Vulkan-Docs/issues/1920 soon (this week probably) and in there I would be more than happy to add some information/column for how many components it is